textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/neextractor/NamedEntityExtractorTest.java

```diff
@@ -0,0 +1,178 @@
+        //TODO: enable test while finish implementation
+       // Assert.assertTrue(contains);
+    }
+
+    /**
+     * Scenario 2: Get next tuple with multiple return
+     * Text: Microsoft, Google and Facebook are organizations
+     */
+    @Test
+    public void getNextTupleTest2() throws Exception {
+        List<ITuple> data = NEExtractorTestConstants.getTest2Tuple();
+        ISourceOperator sourceOperator = getSourceOperator(data.get(0).getSchema(),data);
+
+        List<ITuple> returnedResults = getQueryResults(sourceOperator);
+        List<ITuple> expectedResults = NEExtractorTestConstants.getTest2ResultTuples();
+
+        boolean contains = TestUtils.containsAllResults(expectedResults, returnedResults);
+        //TODO: enable test while finish implementation
+       // Assert.assertTrue(contains);
+
+    }
+
+    /**
+     * Scenario 3: Get next tuple with multiple return with different idendities
+     * Text: Microsoft, Google and Facebook are organizations and Donald Trump and Barack Obama are persons.
+     */
+    @Test
+    public void getNextTupleTest3() throws Exception {
+        List<ITuple> data = NEExtractorTestConstants.getTest3Tuple();
+        ISourceOperator sourceOperator = getSourceOperator(data.get(0).getSchema(),data);
+
+        List<ITuple> returnedResults = getQueryResults(sourceOperator);
+        List<ITuple> expectedResults = NEExtractorTestConstants.getTest3ResultTuples();
+
+        boolean contains = TestUtils.containsAllResults(expectedResults, returnedResults);
+
+        //TODO: enable test while finish implementation
+        //Assert.assertTrue(contains);
+    }
+
+
+
+
+    /**
+     * Scenario 4: Get next tuple with multiple return with different idendities in two sentence(field).
+     * Text: Microsoft, Google and Facebook are organizations and Donald Trump and Barack Obama are persons.
+     */
+    @Test
+    public void getNextTupleTest4() throws Exception {
+        List<ITuple> data = NEExtractorTestConstants.getTest4Tuple();
+        ISourceOperator sourceOperator = getSourceOperator(data.get(0).getSchema(),data);
+
+        List<ITuple> returnedResults = getQueryResults(sourceOperator);
+
+        //TODO:discuss the return type for tuple that have multiple field, how should the span start and end look like?
+        
+        List<ITuple> expectedResults = NEExtractorTestConstants.getTest3ResultTuples();
+
+        boolean contains = TestUtils.containsAllResults(expectedResults, returnedResults);
+
+        //TODO: enable test while finish implementation
+        //Assert.assertTrue(contains);
+    }
+
+
+    /**
+     *
+     * @param schema  The data schema
+     * @param data
+     * @return
+     * @throws Exception
+     *
+     * @about construct a source operator using given schema and data
+     */
+
+    public ISourceOperator getSourceOperator(Schema schema, List<ITuple> data) throws Exception {
+        dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR, schema);
```

Comment:
I didn't know you give your own schema, my bad. I'll see if there's a way this can be added to TestConstants and you can use it in setUp(). For now continue like this! :)